### ISSUE TITLE
Report error for `set -v` used in blocks/conditions

### ISSUE THREAD

**Issue by kkmuffme** [koalaman/shellcheck#3122]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

if [[ "foo" == "$1" ]]
then
        set -v
        echo "hello"
fi
```

#### Here's what shellcheck currently says:
 No issues detected!

#### Here's what I wanted or expected to see:
An error that `set -v` and `set +v` inside blocks do not have any effect, since this is unexpected, as all (?) other "set" work independently of where they're placed
